controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -214,26 +214,30 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	defer func() {
 		// Always attempt to update status.
-		if err := r.updateV1Beta1Status(ctx, controlPlane); err != nil {
-			var connFailure *internal.RemoteClusterConnectionError
+		var connFailure *internal.RemoteClusterConnectionError
+		if err := r.updateStatus(ctx, controlPlane); err != nil {
 			if errors.As(err, &connFailure) {
 				log.Error(err, "Could not connect to workload cluster to fetch status")
 			} else {
-				log.Error(err, "Failed to update KubeadmControlPlane status")
-				reterr = kerrors.NewAggregate([]error{reterr, err})
+				reterr = kerrors.NewAggregate([]error{reterr, errors.Wrap(err, "Failed to update KubeadmControlPlane status")})
 			}
 		}
 
-		r.updateStatus(ctx, controlPlane)
+		if err := r.updateV1Beta1Status(ctx, controlPlane); err != nil {
+			if errors.As(err, &connFailure) {
+				log.Error(err, "Could not connect to workload cluster to fetch status")
```

Comment:
```suggestion
				log.Error(err, "Could not connect to workload cluster to fetch deprecated v1beta1 status")
```
Maybe? Helps us to distinguish if we hit this line or l.220